furbysource.asm

```diff
@@ -506,4 +506,285 @@ InitIO:
+
+;************************************************************
+
+;; IF this is a cold boot , reset command then clear EEPROM and
+;  chose a new name and voice.
+
+Do_cold_boot:
+
+      LDA   #00
+      STA   Warm_cold   ;flag cold boot
+A-22
+      LDA   Stat_0            ;system
+      ORA   #Say_new_name     ;make system say new name
+      STA   Stat_0            ;
+
+;*******  NOTE :::::
+;
+;   VOICE AND NAME SLECTION MUST HAPPEN BEFORE EEPROM WRITE OR
+;   THEY WILL ALWAYS COME UP  00    because ram just got cleared!!!!!!
+
+
+
+; Random voice selection here
+
+      LDA   #80h        ;get random/sequential split
+      STA   IN_DAT            ;save for random routine
+      
+      LDX   #00         ;make sure only gives random
+      LDA   #10h        ;get number of random selections
+      JSR   Ran_seq           ;go get random selection
+      
+      TAX
+      LDA   Voice_table,X     ;get new voice
+      STA   Rvoice            ;set new voice pitch
+
+;**********************************************************************
+*
+
+; On power up or reset, Furby must go select a new name ,,, ahw how
+cute.
+
+      JSR   Random            ;
+      AND   #1Fh        ;get 32 possible
+      STA   Name        ;set new name pointer
+      JSR   Do_EE_write ;write the EEPROM
+
+End_coldinit:
+
+
+
+;ÈÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍ
+;* 'Special initialization prior to normal run mode              *
+;*  Jump to Warm_boot when portD wakes us up
+;ÈÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍÍ
+;
+
+Warm_boot:  ;normal start when Port_D wakes us up.
+
+      JSR   S_EEPROM_READ     ;read data to ram
+
+;Eprom_read_byp:
+
+;***************************************************************
+; If light osc fails, or too dark and that sends us to sleep, we
+; set 'Dark_sleep_prev' and save it in EEPROM in 'Seed_2'.
+; when the sleep routine executes,(00 01 based on this bit)
+; When we wake up we recover this bit and it becomes the previous done
+; flag back in 'Stat_0', so that if the osc is
+A-23
+; still dark or failed, Furby wont go back to sleep.
+      LDA   Seed_2            ;from EEPROM
+      BED   No_prevsleep      ;jump if none
+      LDA   Stat_0            ;system
+      ORA   #Dark_sleep_prev ;prev done
+      STA   Stat_0            ;update
+
+No_prevsleep:
+
+;***************************************************************
+
+      LDA   Spcl_seed1  ;recover start up random number
+      STA   Seed_1            ;set generator
+
+;***************************************************************
+
+
+; Pot_timeL2 is save in ram through sleep mode and then reloaded '
+; Pot_timeL which is the working register for the motor position.
+; This allows startup routines to clear ram without forgetting the
+; last motor position.
+
+      LDA   Pot_timeL2  ;get current count
+      STA   Pot_ imeL   ;save in motor routine counter
```

Comment:
I know the `t` character is missing from `Pot_ imeL` in the PDF, but it may have to be added here just for syntactic correctness.

What do you think - does this kind of fix belong in the layer where we're simply transcribing what's in the PDF, or should there be one layer where the PDF is simply transcribed as faithfully as possible and another layer (maybe a separate script) which fixes up things that are obviously wrong to produce a compile-able ASM file?  I feel like there is a decision to make here, but it's definitely not one I should make on my own without your input.